src/AspNet.Security.OAuth.Autodesk/AutodeskAuthenticationHandler.cs

```diff
@@ -50,6 +51,50 @@ public partial class AutodeskAuthenticationHandler : OAuthHandler<AutodeskAuthen
         return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
     }
 
+    protected override async Task<OAuthTokenResponse> ExchangeCodeAsync(OAuthCodeExchangeContext context)
```

Comment:
The logic used in this overload for PCKE implies that it is used as an alternative to client authentication (since the `client_secret` isn't used when PKCE is). It's traditionally a server implementation mistake as PKCE has never been a way to replace client authentication (both should be used at the same time for a confidential application).

Are we 100% sure Autodesk rejects token requests that specify both PKCE parameters and `client_secret` (in either the form or the `Authorization` header)?

Reply:
Yes, their implementation won't allow both (according to my tests).

Reply:
Hum, I'm intrigued: I just added Autodesk to OpenIddict earlier this week so I gave it a try and the "confidential client" + PKCE combo worked just fine.

Are you sure it's not because you're sending the client credentials twice: once via the `client_id` parameter and once via the `Authorization` header? What kind of client did you create via the Autodesk portal?

Reply:
I create Traditional Web App, and unless Autodesk changed something recently. I recall it didn't work for me when I created the related issue.

Reply:
I just tested my theory and here's the error I get when trying to send the client identifier in both the header and the body (which is illegal):

```json
{
  "error": "invalid_request",
  "error_description": "The 'client_id' is not supported in the request body when Authorization headers are present."
}
```

Try to remove the `tokenRequestParameters.Add("client_id", Options.ClientId);` line and give it another try.